docs_sphinx/advanced/custom_events.rst

```diff
@@ -1,21 +1,78 @@
+with the custom events system in Brian, which is illustrated in this diagram.
+
+.. image:: custom_events.svg
+
+You can see in this diagram that the source `NeuronGroup` has four types
+of events, called ``spike``, ``evt_other``, ``evt_mon`` and ``evt_run``.
+The event ``spike`` is the default event. It is triggered when you
+you include ``threshold='...'`` in a `NeuronGroup`, and has two
+potential effects. Firstly, when the event is triggered it causes the
+reset code to run, specified by ``reset='...'``. Secondly, if there
+are `Synapses` connected, it causes the ``on_pre`` on ``on_post``
+code to run (depending if the `NeuronGroup` is presynaptic or
+postsynaptic for those `Synapses`).
+
+In the diagram though, we have three additional event types. Let's start
+with the first one, ``evt_other``. To understand this, we need to look at
+the `Synapses` object in a bit more detail. A `Synapses` object has
+multiple *pathways* associated to it. By default, there are just two,
+called ``pre`` and ``post``. The ``pre`` pathway is activated by
+presynaptic spikes, and the ``post`` pathway by postsynaptic spikes.
+Specifically, the ``spike`` event on the presynaptic `NeuronGroup` triggers
+the ``pre`` pathway, and the ``spike`` event on the postsynaptic
+`NeuronGroup` triggers the ``post`` pathway. In the example in the diagram,
+we have created a new pathway called ``other``, and the ``evt_other``
+event in the presynaptic `NeuronGroup` triggers this pathway. Note that
+we can arrange this however we want. We could have ``spike`` trigger the
+``other`` pathway if we wanted to, or allow it to trigger both the
+``pre`` and ``other`` pathways. We could also allow ``evt_other`` to
+trigger the ``pre`` pathway. See below for details on the syntax for this.
+
+The third type of event in the example is named ``evt_mon`` and this
+is connected to an `EventMonitor` which works exactly the same way
+as `SpikeMonitor` (which is just an `EventMonitor` attached by default
+to the event ``spike``).
+
+Finally, the fourth type of event in the example is named ``evt_run``,
+and this causes some code to be run in the `NeuronGroup` triggered by
+the event. To add this code, we call `NeuronGroup.run_on_event`. So,
+when you set ``reset='...'``, this is equivalent to calling
+`NeuronGroup.run_on_event` with the ``spike`` event.
+
+Details
+-------
+
+Defining an event
+~~~~~~~~~~~~~~~~~
+
+This can be done with
 the ``events`` keyword in the `NeuronGroup` initializer::
 
     group = NeuronGroup(N, '...', threshold='...', reset='...',
                         events={'custom_event': 'x > x_th'})
 
 In this example, we define an event with the name ``custom_event`` that is
-triggered when the ``x`` variable crosses the threshold ``x_th``. Such events
-can be recorded with an `EventMonitor`::
+triggered when the ``x`` variable crosses the threshold ``x_th``. Note
+that you can define any number of custom events. The dictionary argument
```

Comment:
I find the last sentence ("The dictionary argument is a sequence of ``(event_name, condition)`` pairs.") a bit confusing. Maybe rather something like: "Note that you can define any number of custom events; each event is defined by its name as the key and its condition as the value of the dictionary." (maybe a bit too long-winded)?